openmdao/func_api.py

```diff
@@ -0,0 +1,826 @@
+"""
+API to associate metadata with and retrieve metadata from function objects.
+"""
+
+from numbers import Number
+import ast
+import inspect
+import textwrap
+import warnings
+import numpy as np
+from contextlib import contextmanager
+
+try:
+    import jax
```

Comment:
I think to test certain behaviors, we'll need to add jax to the testing dependencies in setup.py.

and additionally, add some tests were jax is present.

Reply:
I added a couple of tests for the jax based output shape computation.  I'm not sure about making jax a hard testing dependency though.